metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFastTableService.java

```diff
@@ -121,6 +167,10 @@ public HiveConnectorFastTableService(
                 .query(conn, queryBuilder.toString(), handler, params.toArray());
         } catch (SQLException e) {
             throw Throwables.propagate(e);
+        } finally {
+            final long duration = registry.clock().monotonicTime() - start;
+            log.debug("### Time taken to complete getTableNames is {} ms", duration);
+            this.registry.timer(requestTimerId.withTags(tags)).record(duration, TimeUnit.MILLISECONDS);
```

Comment:
I investigated the possibility of using aspects here. The AspectJ can provide the functionality to allow us execute the monitoring code before and after the function call. To get there, we can define the monitor as an aspect, implementing the "advice" code there and defines the joinPoint at each hive function call.  This [article](http://www.concretepage.com/spring/spring-aop-aspectj-annotation-example-with-aspect-pointcut-before-after-around-afterreturning-afterthrowing-advice) has a nice example. 
Spring supports AOP nicely, and most AOP usage is based on Spring. Think we should do it too. 
However, the hive connector is bootstrapped by Guice not by Spring. We need to use the raw AspectJ to write the aspect related codes, which seems a bit painful. At this point, I don't see  much value to use the AspectJ, and converted the hive function names to constants. We can revisit this at some point, and open for discussion.